src/excmds.ts

```diff
@@ -370,6 +370,58 @@ export function loggingsetlevel(logModule: string, level: string) {
 
 // {{{ PAGE CONTEXT
 
+//#content_helper
+export let JUMPLIST = []
+//#content_helper
+export let JUMPCURRENT: number
+//#content_helper
+export let JUMPTIMEOUT: number
+//#content_helper
+export let JUMPED: boolean
+
+//#content
+export function jumpnext(n = 1) {
+    jumpprev(-n)
+}
+
+/** Similar to Pentadactyl or vim's jump list.
+    Should be bound to <C-o> when modifiers are implemented
+*/
+//#content
+export function jumpprev(n = 1) {
+    JUMPCURRENT -= n
+    if (JUMPCURRENT < 0) return back(-JUMPCURRENT)
```

Comment:
If there's no page to go back to, `JUMPCURRENT` gets negative, and you have to jumpnext more times than you'd expect to start going forwards again.

Forwards has a similar problem.

I'd suggest something like this:

```ts
export function jumpprev(n = 1) {
    if (JUMPCURRENT-n < 0) {
        return back(-JUMPCURRENT+n)
    }
    else if (JUMPCURRENT-n >= JUMPLIST.length){
        return forward(JUMPCURRENT - n - JUMPLIST.length + 1)
    }
    JUMPCURRENT -= n
    let p = JUMPLIST[JUMPCURRENT]
    JUMPED = true
    window.scrollTo(p.x, p.y)
}
```